File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        return new int[] { 0, 2, 5, 3, 7, 8, 4 };
    }

    public int Today()
    {
        return this.birdsPerDay[this.birdsPerDay.Length - 1];
    }

    public void IncrementTodaysCount()
    {
        this.birdsPerDay[this.birdsPerDay.Length - 1]++;
    }

    public bool HasDayWithoutBirds()
    {
        int counter = 0;
        for (int i = 0; i < this.birdsPerDay.Length; i++)
            if (this.birdsPerDay[i] <= 0)
                counter++;
        return counter > 0;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int counter = 0;
        for (int i = 0; i < numberOfDays; i++)
            counter += this.birdsPerDay[i];
        return counter;
    }

    public int BusyDays()
    {
        int counter = 0;
        for (int i = 0; i < this.birdsPerDay.Length; i++)
            if (this.birdsPerDay[i] >= 5)
                counter++;
        return counter;
    }
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsPerDay;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        boolean hasDayWithout = false;
        for (int birds : birdsPerDay) {
            if (birds == 0) {
                hasDayWithout = true;
            }
        }
        return hasDayWithout;
    }

    public int getCountForFirstDays(int numberOfDays) {
        if (numberOfDays >= 7) {
            numberOfDays = 7;
        }

        int count = 0;
        for (int i = 0; i < numberOfDays; i++) {
            count += birdsPerDay[i];
        }
        return count;
    }

    public int getBusyDays() {
        int nbBusyDays = 0;
        for (int birds : birdsPerDay) {
            if (birds >= 5) {
                nbBusyDays++;
            }
        }
        return nbBusyDays;
    }
}
